### PR TITLE
Remove useless base classes

### PR DESCRIPTION
**Pull request by ramnes**
`SingleWindow` and `Delegate` are only used by the Slice layout for years, so let's simplify everything and remove them while moving their methods to where they really belong.

### CODE REVIEW

**Comment by tcld** [comment]
For `SingleWindow` I wholeheartedly agree, but I did like the idea of `Delegate` and always saw `Slice` as more of a demo-implementation, just waiting for somebody with some really creative layouting ideas to come along. You are probably right in removing it, however.

**Comment by ramnes** [comment]
Yeah I really don't mind if we revert this commit when needed. Should I make one commit per class?

File: libqtile/layout/slice.py [hunk 1]
```diff
@@ -50,6 +50,18 @@ def remove(self, window):
...
+                screen_rect.x, screen_rect.y,
+                screen_rect.width, screen_rect.height,
+                0,
+                None,
+            )
+            window.unhide()
+        else:
+            window.hide()
+
     def _get_window(self):
```
Comment (tcld) [inline libqtile/layout/slice.py:65]: While you're at it...do we really need this method?

**Comment by tcld** [comment]
One commit per class sounds like a good middleground, I think. Although wouldn't that mean that, if Mr. Layout came along in seven years, somebody would have to remember about that `Delegate`-commit?

I might be overthinking this. I like `Delegate` as a concept, but less code is less code.

**Comment by m-col** [comment]
In my humble opinion we shouldn't keep code 'just in case' nor should we count
on remembering. So +1 to removing any way!

File: libqtile/layout/slice.py (hunk 1 above)
Comment (ramnes) [reply libqtile/layout/slice.py:65]: Nop. :)

**Comment by ramnes** [comment]
Edited the branch to split in two commits and remove the useless _get_window method.

**Comment by tcld** [comment]
Thanks for taking the time, looks good to me. :)

**Pull request merged**